fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneHighlighting.java

```diff
@@ -50,127 +50,214 @@
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.List;
-import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
 
 /**
  * Helper class for highlighting search matches.
  */
 public class LuceneHighlighting {
-    private static final int tokenCountBeforeHighlighted = 3;
-    private static final int tokenCountAfterHighlighted = 3;
-    private static final String highlightedTextConnector = "... ";
+    private static final int defaultSnippetSize = 7;
 
     private LuceneHighlighting() {
     }
 
-    @SuppressWarnings("squid:S3776") // Cognitive complexity is too high. Candidate for later refactoring
-    @Nullable
-    static String searchAllAndHighlight(@Nonnull String fieldName, @Nonnull Analyzer queryAnalyzer, @Nonnull String text,
-                                        @Nonnull Set<String> matchedTokens, @Nonnull Set<String> prefixTokens,
-                                        boolean allMatchingRequired,
-                                        @Nonnull LuceneScanQueryParameters.LuceneQueryHighlightParameters luceneQueryHighlightParameters,
-                                        @Nullable List<Pair<Integer, Integer>> highlightedPositions) {
-        try (TokenStream ts = queryAnalyzer.tokenStream(fieldName, new StringReader(text))) {
-            CharTermAttribute termAtt = ts.addAttribute(CharTermAttribute.class);
-            OffsetAttribute offsetAtt = ts.addAttribute(OffsetAttribute.class);
+    private static class TokenIterator implements Closeable {
+        TokenStream ts;
+        CharTermAttribute termAtt;
+        OffsetAttribute offsetAtt;
```

Comment:
please consider adding `@Nullable`, `@NotNullable` annotation and access modifiers.